__tests__/integration/hathorwallet_others.test.js

```diff
@@ -0,0 +1,1235 @@
+      hWallet1.getAddressAtIndex(4),
+      {
+        token: HATHOR_TOKEN_CONFIG.uid,
+        amount_bigger_than: 2,
+        maximum_amount: 15,
+      }
+    );
+    // FIXME: This result is not consistent, sometimes it fetches only utxo "20".
+    expect(consolidateTx).toStrictEqual({
+      total_utxos_consolidated: 2,
+      total_amount: 9,
+      txId: expect.any(String),
+      utxos: expect.arrayContaining([
+        expect.objectContaining({ amount: 4 }),
+        expect.objectContaining({ amount: 5 }),
+      ])
+    });
+
+    // Validating the updated balance on the wallet
+    await waitForTxReceived(hWallet1, consolidateTx.txId);
+    expect(hWallet1.getAddressInfo(hWallet1.getAddressAtIndex(2)))
+      .toMatchObject({ total_amount_available: 61 });
+    expect(hWallet1.getAddressInfo(hWallet1.getAddressAtIndex(4)))
+      .toMatchObject({ total_amount_available: 9 });
+
+    await waitUntilNextTimestamp(hWallet1, consolidateTx.txId);
+    await cleanWallet1(HATHOR_TOKEN_CONFIG.uid);
+  });
+});
+
+// getAuthorityUtxos acts as a wrapper for selectAuthorityUtxo: testing them together.
+describe('selectAuthorityUtxo and getAuthorityUtxos', () => {
+  /** @type HathorWallet */
+  let hWallet;
+  /** @type string */
+  let tokenHash;
+  beforeAll(async () => {
+    hWallet = await generateWalletHelper();
+  });
+  afterAll(async () => {
+    hWallet.stop();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should work on an empty wallet', async () => {
+    // Default options
+    expect(hWallet.selectAuthorityUtxo(
+      HATHOR_TOKEN_CONFIG.uid,
+      () => true
+    )).toStrictEqual(null);
+
+    // With "many" option
+    expect(hWallet.selectAuthorityUtxo(
+      HATHOR_TOKEN_CONFIG.uid,
+      () => true,
+      { many: true }
+    )).toStrictEqual([]);
+
+    // Testing the wrapper method
+    expect(hWallet.getAuthorityUtxos(fakeTokenUid, 'mint')).toStrictEqual([]);
+    expect(hWallet.getAuthorityUtxos(fakeTokenUid, 'melt')).toStrictEqual([]);
+    expect(() => hWallet.getAuthorityUtxos(fakeTokenUid, 'invalid')).toThrow();
+  });
+
+  it('should find one authority utxo', async () => {
+    // Creating the token
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 1);
+    const { hash: tokenUid } = await createTokenHelper(
+      hWallet,
+      'selectAuthorityUtxo Token',
+      'SAUT',
+      100,
+    );
+    tokenHash = tokenUid;
+
+    // Validating single authority UTXO for a token creation
+    expect(hWallet.selectAuthorityUtxo(tokenHash, wallet.isMintOutput.bind(wallet)))
+      .toStrictEqual([{
+        tx_id: tokenHash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: TOKEN_MINT_MASK,
+      }]);
+    expect(hWallet.selectAuthorityUtxo(tokenHash, wallet.isMeltOutput.bind(wallet)))
+      .toStrictEqual([{
+        tx_id: tokenHash,
+        index: expect.any(Number),
+        address: expect.any(String),
+        authorities: 2,
```

Comment:
Why don't you use `TOKEN_MELT_MASK`?

Reply:
Refactored all uses of these authorities with `1` and `2` to use their respective token masks on [4b175460](https://github.com/HathorNetwork/hathor-wallet-lib/pull/406/commits/4b1754600fe2e6bf913b62737ae522ea1be5d8f5).